buildSrc/src/main/java/org/elasticsearch/test/NamingConventionsCheck.java

```diff
@@ -143,7 +148,7 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
                     String className = filename.substring(0, filename.length() - ".class".length());
                     Class<?> clazz = loadClassWithoutInitializing(packageName + className);
                     if (clazz.getName().endsWith("Tests")) {
-                        if (integTestClass.isAssignableFrom(clazz)) {
+                        if (skipTestsInDisguised == false && integTestClass.isAssignableFrom(clazz)) {
```

Comment:
sure... I see it differently though. we simply don't need to load the class if we don't do any check. We maybe could simply rename the option to "skipIntegTestCheck" or something. let's postpone this though.